src/Saml2/Utils.php

```diff
@@ -397,10 +397,10 @@ public static function post($url, array $parameters = array(), $stay = false)
             $name = urlencode($name);
             if (is_array($value)) {
                 foreach ($value as $v) {
-                    $post["{$name}[]"] = urlencode($v);
```

Comment:
No need to `urlencode()`, the browser will do that itself when the form is submitted. Otherwise, the data will be encoded twice (i.e., `+` => `%2B` => `%252B`).